crates/ruff/src/rules/flake8_pytest_style/rules/assertion.rs

```diff
@@ -248,7 +248,6 @@ pub(crate) fn unittest_assertion(
                     // the assertion is part of a larger expression.
                     if checker.semantic().stmt().is_expr_stmt()
                         && checker.semantic().expr_parent().is_none()
-                        && !checker.semantic().scope().kind.is_lambda()
```

Comment:
Is this change intentional?

Reply:
Yes! Not necessary. See the identical change in `crates/ruff/src/rules/pandas_vet/rules/inplace_argument.rs` that removes the large comment around Case 3. The fact that these tests still pass is a evidence that this change is working as expected.